storm-core/src/clj/backtype/storm/daemon/worker.clj

```diff
@@ -302,9 +301,8 @@
                            port)
                           ]
                          )))
-              (write-locked (:endpoint-socket-lock worker)
-                (reset! (:cached-task->node+port worker)
-                        (HashMap. my-assignment)))
```

Comment:
We should not extend the write lock, we should check tasks->node+port in the read lock when we are sending. See [comment](https://github.com/apache/storm/pull/349#issuecomment-87778672)